src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingContainer.java

```diff
@@ -194,20 +191,25 @@ else if (block instanceof BlockMinecoloniesRack)
      * @param tagName the name of the tag to query
      * @return the BlockPos, or null if not found
      */
-    protected BlockPos getFirstLocationFromTag(String tagName)
+    @Nullable
+    protected BlockPos getFirstLocationFromTag(@NotNull final String tagName)
     {
-        if (tileEntity != null && !tileEntity.getPositionedTags().isEmpty())
+        return getLocationsFromTag(tagName).stream().findFirst().orElse(null);
+    }
+
+    /**
+     * Gets the list of tags, and finds all locations registered there.
+     * @param tagName the name of the tag to query
+     * @return all the matching BlockPos, or an empty list if not found
+     */
+    @NotNull
+    protected List<BlockPos> getLocationsFromTag(@NotNull final String tagName)
+    {
+        if (tileEntity != null)
         {
-            for (final Map.Entry<BlockPos, List<String>> entry : tileEntity.getPositionedTags().entrySet())
-            {
-                if(entry.getValue().contains(tagName))
-                {
-                    //Tagged block for a portal location should be one of the obsidian block at the bottom of the opening
-                    return getPosition().offset(entry.getKey());
```

Comment:
It seems like the nether mine used an offset to get the portal from the "portal" tag.
That seems to be gone during the rewrite

Reply:
That's converting the relative tag offset to world coords.  `getLocationsFromTag` uses `getWorldTagNamePosMap`, which has already applied that offset.

Reply:
Thanks for clarification. I got confused by the comment